synapse/config/jwt_config.py

```diff
@@ -29,11 +29,14 @@ def read_config(self, config, **kwargs):
         jwt_config = config.get("jwt_config", None)
         if jwt_config:
             self.jwt_enabled = jwt_config.get("enabled", False)
-            self.jwt_secret = jwt_config["secret"]
             self.jwt_algorithm = jwt_config["algorithm"]
 
             # The issuer and audiences are optional, if provided, it is asserted
             # that the claims exist on the JWT.
+            self.jwt_secret = jwt_config.get("secret")
+            self.jwt_jwks_uri = jwt_config.get("jwks_uri")
+            self.jwt_subject_claim = jwt_config.get("subject_claim")
+            self.jwt_normalize_user_id = jwt_config.get("normalize_user_id", False)
```

Comment:
I'm assuming we want to default to `False` for backwards compatibility reasons?

Reply:
Exactly.